### PR TITLE
Add some unit tests from cm3leon involving repeat_interleave

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #102733

These actually were fixed by https://github.com/pytorch/pytorch/pull/102570
but that PR doesn't test guard-freeness, so here you go.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

File: test/test_proxy_tensor.py [hunk 1]
```diff
@@ -929,6 +929,35 @@ def f(x):
         self.assertFalse(eval_guards(gm, torch.randn(25, 5)))
         self.assertExpectedInline(show_guards(gm), """L['x'].size()[0] < 20""")
 
+    def test_repeat_interleave(self):
+        def f(src_tokens, beam_size):
+            return src_tokens.repeat_interleave(beam_size, 0)
```
Comment (peterbell10) [inline test/test_proxy_tensor.py:934]: Should we have a test where `beam_size` is symbolic?

File: test/test_proxy_tensor.py [hunk 2]
```diff
@@ -929,6 +929,35 @@ def f(x):
...
+
+        prompt_size = 64
+        vocab_size = 64
+        batch_size = 4
+        src_tokens = torch.randint(1, vocab_size, (batch_size, prompt_size))
+        gm = make_fx(f, tracing_mode="symbolic")(src_tokens, 1)
+        self.assertEqual(len(gm.shape_env.guards), 0)
```
Comment (peterbell10) [inline test/test_proxy_tensor.py:941]: Cool, didn't know it was this easy to inspect the guards.

File: test/test_proxy_tensor.py (hunk 1 above)
Comment (ezyang) [reply test/test_proxy_tensor.py:934]: Sure I can make this symbolic too.

**Review by zou3519 — APPROVED** [review]

**Pull request closed without merging**